src/controller/java/OTAProviderDelegateBridge.cpp

```diff
@@ -0,0 +1,406 @@
+                                                             "(JJ)Lchip/devicecontroller/OTAProviderDelegate$ApplyUpdateResponse;",
+                                                             &handleApplyUpdateRequestMethod);
+    VerifyOrReturn(err == CHIP_NO_ERROR, ChipLogError(Controller, "Could not find handleApplyUpdateRequest method"));
+
+    jobject jResponse = env->CallObjectMethod(otaProviderDelegate, handleApplyUpdateRequestMethod, static_cast<jlong>(nodeId),
+                                              static_cast<jlong>(commandData.newVersion));
+    if (env->ExceptionCheck())
+    {
+        ChipLogError(Support, "Exception in call java method");
+        env->ExceptionDescribe();
+        env->ExceptionClear();
+        return;
+    }
+
+    jmethodID getActionMethod;
+    err = JniReferences::GetInstance().FindMethod(env, jResponse, "getAction", "()I", &getActionMethod);
+    VerifyOrReturn(err == CHIP_NO_ERROR, ChipLogError(Controller, "Could not find getAction method"));
+
+    jmethodID getDelayedActionTimeMethod;
+    err = JniReferences::GetInstance().FindMethod(env, jResponse, "getDelayedActionTime", "()J", &getDelayedActionTimeMethod);
+    VerifyOrReturn(err == CHIP_NO_ERROR, ChipLogError(Controller, "Could not find getDelayedActionTime method"));
+
+    jint jAction             = env->CallIntMethod(jResponse, getActionMethod);
+    jlong jDelayedActionTime = env->CallLongMethod(jResponse, getDelayedActionTimeMethod);
+
+    Commands::ApplyUpdateResponse::Type response;
+    response.action            = static_cast<OTAApplyUpdateAction>(jAction);
+    response.delayedActionTime = static_cast<uint32_t>(jDelayedActionTime);
+    commandObj->AddResponse(cachedCommandPath, response);
+}
+
+void OTAProviderDelegateBridge::HandleNotifyUpdateApplied(CommandHandler * commandObj, const ConcreteCommandPath & commandPath,
+                                                          const NotifyUpdateApplied::DecodableType & commandData)
+{
+    assertChipStackLockedByCurrentThread();
+
+    VerifyOrReturn(mOtaProviderDelegate.HasValidObjectRef(),
+                   commandObj->AddStatus(commandPath, Protocols::InteractionModel::Status::Failure));
+
+    NodeId nodeId = commandObj->GetSubjectDescriptor().subject;
+
+    ConcreteCommandPath cachedCommandPath(commandPath.mEndpointId, commandPath.mClusterId, commandPath.mCommandId);
+
+    JNIEnv * env = JniReferences::GetInstance().GetEnvForCurrentThread();
+
+    jobject otaProviderDelegate = mOtaProviderDelegate.ObjectRef();
+    jmethodID handleNotifyUpdateAppliedMethod;
+    CHIP_ERROR err = JniReferences::GetInstance().FindMethod(env, otaProviderDelegate, "handleNotifyUpdateApplied", "(J)V",
+                                                             &handleNotifyUpdateAppliedMethod);
+    VerifyOrReturn(err == CHIP_NO_ERROR, ChipLogError(Controller, "Could not find handleNotifyUpdateApplied method"));
```

Comment:
command response with failure status?

Reply:
This is simply a Notify and never responds.